cmd/aws-k8s-agent/main.go

```diff
@@ -39,18 +37,17 @@ func _main() int {
 
 	log.Infof("Starting L-IPAMD %s  ...", version)
 
-	kubeClient, err := k8sapi.CreateKubeClient()
-	if err != nil {
-		log.Errorf("Failed to create client: %v", err)
+	//Check API Server Connectivity
+	if k8sapi.CheckAPIServerConnectivity() != nil{
 		return 1
 	}
 
-	eniConfigController := eniconfig.NewENIConfigController()
-	if ipamd.UseCustomNetworkCfg() {
-		go eniConfigController.Start()
+	standaloneK8SClient, k8sClient, err := k8sapi.CreateKubeClients()
```

Comment:
Not a fan of this "return multiple magic clients" API :(
Perhaps break it into a regular "create a client" function (ie: the old `CreateKubeClient`) and then a _separate_ function that creates a caching client from a 'raw' client.  Then it will be clear to the caller (here) what the two are, and it will also be obvious for the cases where you only want one of those clients.

Reply:
Will split these in to separate functions..